ocp_utilities/infra.py

```diff
@@ -49,6 +53,30 @@ def assert_nodes_schedulable(nodes):
         )
 
 
+def assert_pods_failed_or_pending(pods: list) -> None:
+    """
+    Validates all pods and flag failed or pending pods
+
+    Args:
+         pods: List of pod objects
+
+    Raises:
+        PodsFailedOrPendingError, if there are failed or pending pods
+    """
+    LOGGER.info("Verify all pods are not failed nor pending.")
+
+    failed_or_pending_pods = []
```

Comment:
I don't see where you append the pods to this list.

Reply:
Done.